api/src/main/java/org/openmrs/validator/UserValidator.java

```diff
@@ -125,12 +125,14 @@ public void validate(Object obj, Errors errors) {
 				}
 			}
 			
+			
 			if (!StringUtils.isBlank(user.getEmail())) {
```

Comment:
@sacull  @dkayiwa  so i think you followed this commit, me trying to change the structure of these if statements line 129 breaks the testing in UserValidatorTest. My first approach on this was TODO  like
  **if((!StringUtils.isBlank(user.getEmails()) ||(!isEmailValid(user.getEmail())) ){
}**
but that lead to breakage of testing, i didnt wanted to go away from refactoring from this

Reply:
Why || not &&?

Reply:
Both conditions must be met, not just one of them. ;)

Reply:
That method using && block tests let me use asap and you have a look

Reply:
I already know where the problem is, you have too many parentheses and just one of you moved at the end instead of being in the middle.

Take a look at this entry and consider why it will work and how it is different from yours: 
`if (!StringUtils.isBlank(user.getEmail()) && !isEmailValid(user.getEmail())) {}`

Reply:
One more thing, if you want to continue this PR, remember to assign a JIRA ticket to yourself.

Reply:
True dat @sacull Actually with that matter we need to add an else close to get rid of NPE if we just used it like that we would fall in NPE  which tends to happen in UserValidatorTest. take a look of how it looks now